code/modules/projectiles/gun.dm

```diff
@@ -35,6 +35,7 @@
 	var/semicd = 0						//cooldown handler
 	var/weapon_weight = WEAPON_LIGHT
 	var/list/restricted_species
+	var/bolt_open = FALSE
```

Comment:
I'm not sure it is a good idea to add this variable to the base class. Usually, a good rule of thumb is to not make something generic if it won't be required as such for a while.

My suggestion would be to put this in the mosin definition itself and override `update_icon_state` in accordance with the behaviour you'd expect.

Reply:
Thanks for the review. I will take a look and deconflict this PR as soon as I have time.